src/e2eTest/utils/actions/custom-actions/createCase.action.ts

```diff
@@ -0,0 +1,85 @@
+import Axios, {AxiosInstance, AxiosResponse} from 'axios';
+import {TestConfig} from 'config/test.config';
+import {
+  getIdamAuthToken,
+  getServiceAuthToken
+} from '../../helpers/idam-helpers/idam.helper';
+import {actionData, IAction} from '../../interfaces/action.interface';
+import {Page} from '@playwright/test';
+
+let caseInfo: { id: string; fid: string, state: string };
+const testConfig = TestConfig.ccdCase;
+
+export class CreateCaseAction implements IAction {
+  private eventToken?: string;
+
+  constructor(
+    private readonly axios: AxiosInstance,
+  ) {
+  }
+
+  async execute(page: Page, fieldName?: actionData, value?: actionData): Promise<void> {
+    if (!fieldName) throw new Error('Missing fieldName');
+    const caseApiInstance = caseApi();
+    caseInfo = await caseApiInstance.createCase(fieldName);
+  }
+
+  async getEventToken(): Promise<string> {
+    if (!this.eventToken) {
+      const tokenResponse: AxiosResponse<{ token: string }> = await this.axios.get(
+        `/case-types/${testConfig.caseType}/event-triggers/${testConfig.eventName}`
+      );
+      this.eventToken = tokenResponse.data.token;
+    }
+    return this.eventToken;
+  }
+
+  async createCase(caseData: actionData): Promise<{ id: string; fid: string; state: string }> {
+    const eventToken = await this.getEventToken();
+    const event = {id: `${testConfig.eventName}`};
+    const payloadData = typeof caseData === 'object' && 'data' in caseData ? caseData.data : caseData;
+    try {
+      const response = await this.axios.post(
+        `/case-types/${testConfig.caseType}/cases`,
+        {
+          data: payloadData,
+          event: event,
+          event_token: eventToken,
+        }
+      );
+      return {
+        id: response.data.id,
+        fid: formatCaseNumber(response.data.id),
+        state: response.data.state,
+      };
+    } catch (err) {
+      throw err
+      throw new Error('Case could not be created.');
+    }
+  }
+}
+
+export const caseApi = (): CreateCaseAction => {
```

Comment:
Minor, can you rename caseApi to dataStoreApi, please? So it is clear which service this is referring to.